crates/nu-protocol/src/value/mod.rs

```diff
@@ -3445,6 +3445,13 @@ impl Value {
                     Err(ShellError::DivisionByZero { span: op })
                 }
             }
+            (Value::Duration { val: lhs, .. }, Value::Duration { val: rhs, .. }) => {
+                if *rhs != 0 {
+                    Ok(Value::duration(lhs % rhs, span))
```

Comment:
Should be `Value::Int()`, not duration.

The time units cancel out in this division-like operation.

See `src/tests/test_math.rs` for a place to implement some unit tests.